### PR TITLE
examples: convert configurations to v2 format

### LINKED ISSUE #2139: configs/examples: convert all to v2

### PR DESCRIPTION
**Pull request by mccv**
examples: v2 configs

Description:
Convert all example configs to v2 format.
Legacy configs are stored in a `deprecated_v1` directory in each example. They are still verified during bazel config tests.

Risk Level: Low

Testing:

Verified all examples using docker-compose on my local machine. Note that there is an existing mismatch with docs (which have links to localhost) and using docker-machine (which creates a distinct IP).

Also ran `bazel test //test/config_test/...`

Release Notes:

Example configs migrated to the v2 format.

### CODE REVIEW

**Comment by mattklein123** [comment]
Thank you @mccv!!! 💯 💯 💯 

A few quick comments:
1) Can we remove the "fixes" from the PR description as we still need to convert all the configs in the "configs" folder (but this is great progress!!!). 
2) I would prefer that we keep the v1 configs around for testing purposes until we fully delete v1. Can we potentially move all of these configs into a v1_deprecated folder or something and then make sure they still get tested as part of the config tests? (That would involve updating the BUILD file to include both v1 and v2 configs and updating https://github.com/envoyproxy/envoy/blob/master/test/config_test/example_configs_test.cc#L11 to have the right number.

Thank you!!!

**Comment by mccv** [comment]
Sounds good on all notes. I'm stepping out for a bit, but should have an updated version later today.

**Comment by mccv** [comment]
Ok, updated based on feedback. Note that the v1 configs get a new directory _and_ a new name, because the test step just mashes all the example configs into a single tar file.

**Comment by mattklein123** [comment]
Sweet! @mccv can you fix format/DCO also?

**Comment by mccv** [comment]
Done, sorry for not reading to the bottom of contributing.md

**Comment by mccv** [comment]
And... fixed formatting in the BUILD file

File: examples/jaeger-tracing/front-envoy-jaeger.json [hunk 1]
```diff
@@ -1,84 +1,107 @@
...
-                {
-                  "name": "backend",
-                  "domains": ["*"],
-                  "routes": [
-                    {
-                      "prefix": "/",
-                      "cluster": "service1",
-                      "decorator": {
-                        "operation": "checkAvailability"
+  "static_resources": {
```
Comment (htuch) [inline examples/jaeger-tracing/front-envoy-jaeger.json]: QQ: do we want to switch to YAML as well for human readable examples when doing the v2 conversion?

Comment (mattklein123) [reply examples/jaeger-tracing/front-envoy-jaeger.json]: I would prefer YAML, but I didn't want to ask @mccv to do that extra work. Is there a quick way to convert JSON to YAML via tool?

**Comment by htuch** [comment]
https://www.json2yaml.com/ ? :) I imagine there are a bunch of others.

**Comment by mattklein123** [comment]
> https://www.json2yaml.com/ ? :) I imagine there are a bunch of others.

Aha. Cool. Yes let's convert. @mccv do you mind?

**Comment by mccv** [comment]
yeah, no problem. Do you want to keep the json files alongside as a reference, and just add yaml files (and use them as the config source in the docker-compose files?)

**Comment by mattklein123** [comment]
My preference would be to just drop JSON in v2 and use YAML for the config source, etc.

**Comment by mccv** [comment]
Alright, slightly more involved than I'd thought, but converted them all and ran through the tests and sandbox use cases again

File: examples/front-proxy/front-envoy.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (htuch) [inline examples/front-proxy/front-envoy.yaml]: What is this leading `---` for?

Comment (mccv) [reply examples/front-proxy/front-envoy.yaml]: added by the converter, and apparently indicates "beginning of file". Probably harmless to remove them if it's irksome

Comment (mattklein123) [reply examples/front-proxy/front-envoy.yaml]: My mild preference is to remove them if you don't mind just because I think many people will be confused.

**Comment by mccv** [comment]
done

**Pull request merged**